tests/test_audio_signal.py

```diff
@@ -38,22 +38,39 @@ def test_signal_init_assertions():
     with pytest.raises(ValueError, match="Invalid domain"):
         Signal(1, 44100, domain="space")
 
+    with pytest.raises(ValueError, match="n_samples can not be larger"):
+        Signal(1, 44100, domain="freq", complex=True, n_samples=10)
+
 
 def test_signal_init_time_dtype():
     """
     Test casting and assertions of dtype (also test time setter because
     it is called during initialization)
     """
-    # integer to float casting
+    # pass integer values, expect float casting
     signal = Signal([1, 2, 3], 44100)
     assert signal.time.dtype.kind == "f"
 
-    # float
+    # pass real-valued int data and set complex flag, expect complex casting
+    signal = Signal([1, 2, 3], 44100, complex=True)
+    assert signal.time.dtype.kind == "c"
+
+    # pass float data
     signal = Signal([1., 2., 3.], 44100)
     assert signal.time.dtype.kind == "f"
 
-    # complex
-    with pytest.raises(ValueError, match="time data is complex"):
+    # pass real-valued float data and set complex flag
+    signal = Signal([1., 2., 3.], 44100, complex=True)
+    assert signal.time.dtype.kind == "c"
+
+    # pass complex-valued data and set complex flag
+    signal = Signal([1+1j, 2+2j, 3+3j], 44100, complex=True)
+    assert signal.time.dtype.kind == "c"
+
+    # pass complex data but dont set complex flag
+    with pytest.raises(ValueError, match="time data is complex, "
+                                         "set complex flag or pass "
+                                         "real-valued data."):
         Signal([1+1j, 2+2j, 3+3j], 44100)
```

Comment:
should we move this up to test_signal_init_assertions?

Reply:
Good point, done so.